### PR TITLE
Enhance Compatibility for Custom Featured Image & Category Metadata & tags

### PR DESCRIPTION
**Pull request by bafi**
## Description

- When publishing or updating content via PCC, the system will now include custom metadata labeled `Categories`, `FeaturedImage` and metadata for the Yoast plugin (`title` and `description`). These will be correctly mapped in WordPress, with `Categories` being applied as WordPress categories, `FeaturedImage` being set as the post's featured image (imported into the WordPress Media Library), and the Yoast `title` and `description` being updated if these fields are *set* in PCC.

## Testing instructions

1. **Test 1:** Validate that the custom “Categories” metadata is correctly fetched from PCC during content synchronization.
2. **Test 2:** Confirm that the custom “Categories” metadata is accurately mapped and displayed as WordPress categories when content is published or updated.
3. **Test 3:** Validate that the custom “FeaturedImage” metadata is correctly fetched from PCC during content synchronization.
4. **Test 4:** Confirm that the custom “FeaturedImage” metadata is imported into the WordPress Media Library and set as the post's featured image.
5. **Test 5:** Validate that the Yoast title and description are updated only if the corresponding fields in PCC are not set.
6. **Test 6:** Ensure that if a category, featured image already exists, the existing data is reused without duplication.
7. **Test 7:** Ensure that if the custom “FeaturedImage,” “Categories,” or Yoast metadata is missing, these fields remain as it is in WordPress.

## Checklist:

- [X] I've tested the code.
- [X] My code follows the repository code
  style. 
- [X] My code follows the accessibility
  standards. 
- [X] My code follows the PHP DocBlock documentation
  standards.

### CODE REVIEW

File: app/PccSyncManager.php [hunk 1]
```diff
@@ -113,17 +122,173 @@ private function createOrUpdatePost($postId, Article $article, bool $isDraft = f
 			'post_name' => $article->slug,
 			'post_type' => $this->getIntegrationPostType(),
 		];
+		// Set post excerpt if description is available.
```
Comment (al-esc) [inline app/PccSyncManager.php:125]: here, the `$data` array should be built with the metadata title, if it exists:

```
'post_title' => $article->metadata['title'] ?? $article->title,
```

File: app/PccSyncManager.php [hunk 2]
```diff
@@ -113,17 +122,173 @@ private function createOrUpdatePost($postId, Article $article, bool $isDraft = f
...
+	{
+		if (!isset($article->metadata['FeaturedImage'])) {
+			return;
+		}
+		// If the feature image is empty, delete the existing thumbnail.
+		$featuredImageURL = $article->metadata['FeaturedImage'] ?? '';
+		if ('' === $featuredImageURL) {
+			delete_post_thumbnail($postId);
+			return;
+		}
```
Comment (al-esc) [inline app/PccSyncManager.php:192]: remove these lines;
we don't want to remove the featured image from the WP article if the featured image is missing from the PCC article metadata

File: app/PccSyncManager.php [hunk 3]
```diff
@@ -39,7 +39,16 @@ public function fetchAndStoreDocument(
...
 			$documentId,
-			[],
+			[
+				'id',
+				'slug',
+				'title',
+				'siteId',
+				'tags',
+				'content',
+				'snippet',
```
Comment (al-esc) [inline app/PccSyncManager.php:49]: `snippet` field is not used, can be removed

File: app/PccSyncManager.php [hunk 4]
```diff
@@ -39,7 +39,16 @@ public function fetchAndStoreDocument(
 		$articlesApi = new ArticlesApi($this->pccClient());
 		$article = $articlesApi->getArticleById(
 			$documentId,
-			[],
+			[
+				'id',
+				'slug',
+				'title',
+				'siteId',
```
Comment (al-esc) [inline app/PccSyncManager.php:46]: `siteId` field is not used, can be removed

**Review by al-esc — CHANGES_REQUESTED** [review]
couple comments before we can merge

File: app/PccSyncManager.php (hunk 2 above)
Comment (bafi) [reply app/PccSyncManager.php]: As discussed, We will keep it as it is and clean up code

**Review by al-esc — APPROVED** [review]

**Pull request merged**